plugins/filter/accumulate.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) Max Gautier <mg@max.gautier.name>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+DOCUMENTATION = '''
+  name: accumulate
+  short_description: Produce a list of accumulated sums of the input list contents
+  version_added: 10.1.0
+  author: Max Gautier (@VannTen)
+  description:
+    - Passthrough to the L(Python itertools.accumulate function,https://docs.python.org/3/library/itertools.html#itertools.accumulate).
+    - Transforms an input list into the cumulative list of results from applying addition to the elements of the input list.
+    - Addition means the default Python implementation of C(+) for input list elements type.
+  options:
+    _input:
+      description: A list.
+      type: list
+      elements: any
+      required: true
+'''
+
+RETURN = '''
+  _value:
+    description: A list of cumulated sums of the elements of the input list.
+    type: list
+    elements: any
+'''
+
+EXAMPLES = '''
+- name: Enumerate parent directories of some path
+  ansible.builtin.debug:
+    var: >
+       "/some/path/to/my/file"
+       | split('/') | map('split', '/')
+       | community.general.accumulate | map('join', '/')
+    # Produces: ['', '/some', '/some/path', '/some/path/to', '/some/path/to/my', '/some/path/to/my/file']
+'''
+
+from itertools import accumulate
+
+
+def list_accumulate(sequence):
+    return accumulate(sequence)
+
+
+class FilterModule(object):
+
+    def filters(self):
+        return {
+            'accumulate': list_accumulate,
```

Comment:
Why not

```suggestion
            'accumulate': accumulate,
```
?

`list_accumulate` is not really doing anything.

Reply:
It is doing something: it's allowing exactly one positional argument, this restricting the arguments to what is documented.

(It also allows to add better error checking at some point.)

Reply:
fair enough

Reply:
Yeah I had considered whether to forward more arguments to python accumulate, but:
- I don't think there is an ergonomic way to pass a binary function
- Passing `initial` would be roughly equivalent to `([initial_elem] + input_list) | accumulate)` so I don't think it's worth adding an optional argument.

> (It also allows to add better error checking at some point.)

What do you have in mind ? I could just as well add it now.

Reply:
Somethings like checking for args and kward and raising an exception in there is some ?

Reply:
> What do you have in mind ? I could just as well add it now.

Checking whether the `sequence` is actually a sequence.

With `from collections.abc import Sequence` you can use `if not isinstance(sequence, Sequence):` to test it.

(This also accepts strings as sequences. That's generally not so great, but here strings are also OK, I guess.)

Reply:
Do you have time to work on this before say in a week?

Reply:
Also, it appears that checking for Sequence does not catch subtler invalid case, such as non compatible list members :
```
TASK [filter_accumulate : Test accumulate filter] ******************************
fatal: [testhost]: FAILED! => {"msg": "The conditional check '['ac', 1] | community.general.accumulate == ['ac', 'ac1']' failed. The error was: Unexpected templating type error occurred on ({% if ['ac', 1] | community.general.accumulate == ['ac', 'ac1'] %} True {% else %} False {% endif %}): can only concatenate str (not \"int\") to str. can only concatenate str (not \"int\") to str"}
```

Reply:
> -> this one errors when checking for a Sequence type.

If you really want to do that, you can always use the `| list` filter to convert the dict to a list (of its keys) first.

> Also, it appears that checking for Sequence does not catch subtler invalid case, such as non compatible list members :

Obviously it cannot catch all invalid input, but it can catch some of the more obvious ones.

> Wdyt ? I would tend to got the more permissive way, and document the unobvious cases, but I don't have strong opinions.

I would definitely check for `Sequence`.